service/application.go

```diff
@@ -169,6 +180,15 @@ func New(params Parameters) (*Application, error) {
 	return app, nil
 }
 
+// Run starts the collector according to the command and configuration
+// given by the user, and waits for it to complete.
+func (app *Application) Run() error {
```

Comment:
Moved this one from the bottom of the file since it was the only public function not on top.